functions/src/triggers/storage.ts

```diff
@@ -0,0 +1,45 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import { ObjectMetadata } from 'firebase-functions/lib/providers/storage';
+import * as sharp from 'sharp';
+
+import { ImageData } from '../../../models/image-data.models';
+
+export const convertToWebP = functions.storage.object().onFinalize(async (object: ObjectMetadata) => {
+  const image = new ImageData(object);
+
+  if (!image.contentType.startsWith('image/')) return null;
+  if (image.filePath.endsWith('.webp')) return null;
+
+  const bucket = admin.storage().bucket(image.bucketName);
+
+  const [fileBuffer] = await bucket.file(image.filePath).download();
+
+  await convertAndResizeImage(fileBuffer, image, bucket);
+
+  return null;
+})
+
+async function convertAndResizeImage(fileBuffer, image, bucket) {
```

Comment:
Doesn't seem like this function is abstracting much complexity as it is declared directly under its only usage.

Reply:
Changes are in 6a28be9